src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -704,6 +738,10 @@ public bool DeleteCollection(DocumentsOperationContext context, string name, Lis
                 IsSystemDocument = isSystemDocument,
             });
 
+            var vectorValue = GetChangeVectorForCurrentDb(context);
```

Comment:
You are setting the GLOBAL value here, but this is never correct.
The global change vector is only set on replication, with the local value being added on the fly from our most recent etag.

You are also not actually adding the data to the current document.